courses/api.py

```diff
@@ -345,3 +345,16 @@ def defer_enrollment(
     except EdxEnrollmentCreateError:  # noqa: TRY302
         raise
     return from_enrollment, first_or_none(to_enrollments)
+
+
+def generate_course_readable_id(course_tag):
```

Comment:
> I think we should do one of the following:
> 
> The method name should mention emeritus to explicitly portray that this is meant for emeritus
> We should have something like emeritus_api.py where all the related methods that are only meant for emeritus should be moved.

This is a generic method to generate the readable IDs for the courses. It is not linked with Emeritus, We can use any course tag and it will return the formatted readable ID in the format `course-v1:xPRO+<COURSE_TAG>`

I do agree with your comment to move all the Emeritus API methods to emeritus_api.py

> Should this function throw an exception if the course_tag passed is empty or null?

It should throw an exception. I will add that.